user/src/components/RegisterCarousel/useRegistration.ts

```diff
@@ -38,72 +83,112 @@ const determineErrorStep = (
  * - 登録成功時の処理
  * - ローディング状態の管理
  */
-export const useRegistration = (
-  validateForm: (step: number) => Promise<boolean>,
-  goToStep: (step: number) => void,
-  currentStep: number,
-  onClose?: () => void
-) => {
-  const { register: authRegister, isLoading } = useAuth();
-  const [displayError, setDisplayError] = useState<string | null>(null);
+export function useRegistration(
+  validateForm: (step: number) => Promise<boolean>, // 各ステップのバリデーション関数
+  goToStep: (step: number) => void, // 指定ステップに移動する関数
+  currentStep: number, // 現在のステップ番号
+  handleSubmit: UseFormHandleSubmit<RegisterFormSchema> // フォーム送信ハンドラー
+) {
+  const [isLoading, setIsLoading] = useState(false); // ローディング状態
+  const [displayError, setDisplayError] = useState<string>(); // 表示するエラーメッセージ
+  const router = useRouter(); // ルーターオブジェクト
 
   /**
    * APIエラー発生時に対応するステップに移動
-   * @param apiErrors APIから返されたエラー情報
+   * @param errors APIから返されたエラー情報
    */
-  const navigateToErrorStep = useCallback(
-    (apiErrors: Record<string, string[]> | undefined) => {
-      const errorStep = determineErrorStep(apiErrors);
-      if (errorStep >= 0) {
-        goToStep(errorStep);
+  const navigateToStep = useCallback(
+    (errors?: ApiErrors) => {
+      const step = detectErrorStep(errors); // エラーが発生したステップを特定
+      if (step >= 0) {
+        goToStep(step); // 該当ステップに移動
```

Comment:
`detectErrorStep()` っていう関数名があるんで `step` は `errorStep` に揃えたほうが分かりやすいかも

```suggestion
  const navigateToErrorStep = useCallback(
    (errors?: ApiErrors) => {
      const errorStep = detectErrorStep(errors); // エラーが発生したステップを特定
      if (errorStep >= 0) {
        goToErrorStep(errorStep); // 該当ステップに移動
```